lib/lib-upload/src/main/java/org/sagebionetworks/upload/multipart/S3MultipartUploadDAOImpl.java

```diff
@@ -110,6 +224,11 @@ public void validateAndAddPart(AddPartRequest request) {
 		s3Client.deleteObject(request.getBucket(), request.getPartKey());
 	}
 
+	@Override
+	public void validatePartCopy(CompositeMultipartUploadStatus status, long partNumber, String partMD5Hex) {
+		// Nothing to validate

```

Comment:
We need to validate with AWS support if the copy guarantees integrity. If not we can potentially open a stream to the source object for the range of bytes to compute the MD5 and compare it with the added one (not ideal but still better than re-uploading).